addons/drawbuild/functions/fnc_drawBuildZeus.sqf

```diff
@@ -14,92 +14,88 @@ Starts the selection handler to select multiple points for you to draw
-
-	// put in object
-	private _spot = IDC_ICON_GRID_FIRST + _forEachIndex;
-	// get spot
-	private _ctrlSpot = _display displayCtrl _spot;
-
-	// set picture and tooltip
-	_ctrlSpot ctrlSetText _picture;
-	_ctrlSpot ctrlSetTooltip _tooltip;
-
-	// handler for click on icon
-	[_ctrlSpot, "ButtonClick", {
-            params ["_ctrlSpot"];
-            _thisArgs params ["_object", "_display"];
+// TODO: sort by display name? Or by category?
+private _drawPresets = [keys GVAR(drawBuildPresets), [], { getText(configfile >> "CfgVehicles" >> _x >> "displayName") }] call BIS_fnc_sortBy;
 
-			// close dialog
-			closeDialog 1;
-
-			// get options for simulation and damage 
-			private _ctrlSimCheckbox = _display displayCtrl IDC_CHECKBOX_SIMULATION;
-			private _ctrlDmgCheckbox = _display displayCtrl IDC_CHECKBOX_DAMAGE;
-
-			private _enableSim = cbChecked _ctrlSimCheckbox;
-			private _enableDmg = cbChecked _ctrlDmgCheckbox;
-
-			// start draw building
-			[_object, _enableSim, _enableDmg] call FUNC(drawBuildSelectPosition);
-			
-    }, [_mainItem, _display]] call CBA_fnc_addBISEventHandler;
-
-} forEach _arrOptions;
+private _objects = [];
+private _prettyNames = [];
+{
+    private _displayName = getText(configfile >> "CfgVehicles" >> _x >> "displayName");
+    private _picture = getText(configfile >> "CfgVehicles" >> _x >> "editorPreview");
+
+    _objects pushBack _x;
+    _prettyNames pushBack [_displayName, "", _picture];
+
+} forEach _drawPresets;
+
+[
+	"Draw Build", 
+	[
+		["EDIT","Filter",["", {
+
+			private _filter = _this;
+			private _filteredObjects = (keys GVAR(drawBuildPresets) select { ([_filter, getText(configfile >> "CfgVehicles" >> _x >> "displayName")] call BIS_fnc_inString) });
+			_filteredObjects = [_filteredObjects, [], { getText(configfile >> "CfgVehicles" >> _x >> "displayName") }] call BIS_fnc_sortBy;
+
+			private _listbox = (allControls (uiNamespace getVariable "zen_common_display")) select { ctrlType _x == 5} select 0;
+			lbClear _listbox;
+			private _index = 0;
+			{
+				_listbox lbAdd getText(configfile >> "CfgVehicles" >> _x >> "displayName");
+			    _listbox lbSetPicture [_index, getText(configfile >> "CfgVehicles" >> _x >> "editorPreview")];
+			    _listbox setVariable [str _index, _x];
+			    _index = _index+1;
+			} forEach _filteredObjects;
+
+
+			// TODO: this duplicates the code below - move to a function?
+			// There's got to be a better way than this
+			private _edit = (allControls (uiNamespace getVariable "zen_common_display")) select { ctrlType _x == 2} select 2;
+
+			// Surely a better way to do this too
+			((uiNamespace getVariable "zen_common_display") getVariable "zen_dialog_params") params ["_controls", "_onConfirm", "_onCancel", "_args", "_saveID"];
+			private _values = _controls apply {
+			    _x params ["_controlsGroup", "_settings"];
+			    [_controlsGroup, _settings] call (_controlsGroup getVariable "zen_dialog_fnc_value")
+			};
```

Comment:
Not sure I understand this part enough to be able to suggest better ways. Are you targeting the named components that ZEN makes to be able to modify it?

Reply:
Yes - grabs the third edit box in the UI, then fills that based on the values (stored in ZEN) of the lb component. Might be a "cleaner" way to do it, but it works at least!